### DISCUSSION TITLE
Tenant Permission Middleware

### DISCUSSION THREAD

**Discussion by chrisgrande** [django-tenants/django-tenants#570] [Q&A]
The project I'm working on has mostly all tenant pages public facing except for some admin dashboard pages, where one user could have access across multiple tenants.

To control access I added an `admins` `ManyToManyField` to the the `tenant` model along with some middleware that performs:

```
if request.user.is_authenticated:
    if request.user.is_superuser:
        pass
    else:
        if request.user in request.tenant.admins.all():
            pass
        else:
            logout(request)
```

Which just logs someone out if they try to access a dashboard page of a tenant they aren't an admin of along with allowing superusers to access all dashboards. This also allows for a main login page that once they login routes them to a list of all their tenants if they are an admin of more than one.

Am I missing something with the simplicity of this?

**Comment by mronoffon** [comment]
you may want to look into django-tenant-users

**Reply by chrisgrande** [reply]
I forgot to write that had — it hasn't yet been updated for Django 3, which lead me to this simpler solution.

**Reply by tomturner** [reply]
I believe django-tenant-users will now work django 3
